packages/cypress/src/steps.ts

```diff
@@ -281,10 +281,13 @@ function groupStepsByTest(tests: Test[], steps: StepEvent[]): Test[] {
         "beforeAll",
       ];
       for (const phase of phases) {
-        const stepWithError = t.events[phase].reverse().find(t => t.data.error);
-        if (stepWithError) {
-          t.error = stepWithError.data.error;
-          break;
```

Comment:
`Array.prototype.reverse` mutates the events array in place. This is _not_ what we want.

Furthermore it's kind of wasteful to call this method anyway. We could use `Array.prototype.findLast` except I'm not sure what our minimum Node target is so...